src/typecheck.rs

```diff
@@ -52,13 +52,14 @@ impl From<lang::Expr> for operator::Expr {
     }
 }
 
-const DEFAULT_LIMIT: f64 = 10.0;
+const DEFAULT_LIMIT: i64 = 10;
 
 impl lang::InlineOperator {
     /// Convert the operator syntax to a builder that can instantiate an operator for the
     /// pipeline.  Any semantic errors in the operator syntax should be detected here.
     pub fn semantic_analysis(
         self,
+        pipeline: &lang::QueryContainer,
```

Comment:
having this be `QueryContainer` is a bit confusing and doesn't really get at the intention. If it doesn't lead to too much of a headache, can you split the error reporting bits into their own trait and pass that instead?

Reply:
Added an `ErrorBuilder` trait defined in the new errors.rs module and pass that around.

(I'm not big on the ErrorBuilder/ErrorReporter names...)

Reply:
Yeah no worries, I may got through an do some renaming at some point if I come up with better names